pkg/lightning/backend/local.go

```diff
@@ -1884,7 +1883,24 @@ func (w *LocalWriter) writeKVsOrIngest(desc localIngestDescription) error {
 		}
 	}
 
-	// if write failed only because of unorderedness, we immediately ingest the memcache.
+	// if write failed because of unorderedness, we immediately ingest or write the memcache.
```

Comment:
Since we have disabled the implicit compaction, the main difference between ingest and write-batch approach is the memcache size. For ingest, it's 128MB; and For write-batch it's 512MB. So ingest approach will generated 3 times more sst files than wb.  This is likely a temporary solution. Maybe a simpler way is to use ingest for data engine and write batch for index engine. 

Because the write-base approach is proved to be very slow, especially when there is only one table will a lot of indexes. So I don't think this change can address the performance problem. Shall we wait for #753 with the manual compaction?

Reply:
> Maybe a simpler way is to use ingest for data engine and write batch for index engine.

so you mean just remove the overlap check?

> So I don't think this change can address the performance problem.

this PR addresses the problem that (during Disk Quota) FlushAllEngines stuck because of having 17000 L0 files. i doubt #753 has tested that scenario.

Reply:
But the write-batch approach still have this problem, but because it's batch-size is larger, it this sceniro, the L0 SST files should be 17000 / 4 , this number is still very big. Without compaction (either manually or implicit) we can't fully address this problem. And the write-batch is slow due to skip-list insert. 

We can accept this pr, but this is a temp solution. #753 uses manual compaction to control the SST file number. With 10TB source files size, it can ensure the total SST files are no more than 1k.

Reply:
BTY, I'm afraid this approach will introduce performance regression because for each write-batch it needs to first check the db iterator and then fall back to the write-batch solution if the overlap is somewhat severe

Reply:
If we write all data by writebatch in index-engine, we may be blocked by flush-thread because all data must be flushed by only  thread even if there are big data writing into index engine.

Reply:
Maybe we could split data of source into more than 4 index-engine and just merge all result of them by a merge iterator?